### DISCUSSION TITLE
Pagination links on groups ?

### DISCUSSION THREAD

**Discussion by gotson** [opds-community/drafts#68] [Q&A] [answered]
I am implementing OPDS2 for [Komga](https://komga.org/), and i was wondering if [Pagination links](https://drafts.opds.io/opds-2.0.html#3-pagination) can be added in [Groups](https://drafts.opds.io/opds-2.0.html#15-groups) ?

If i have a feed containing 2 groups, can both those groups contain pagination links ?

In which case, is the OPDS reader expected to allow navigation within the pagination in those 2 separate groups ?

For example with Thorium Reader, if the 2 groups _Main menu_ and _Libraries_ contained pagination links, should Thorium allow navigation with prev/next for both those sections ?

<img width="1439" alt="image" src="https://github.com/opds-community/drafts/assets/2139133/eb76f268-ae87-4096-b7ad-9d3d757d926f">

**Comment by HadrienGardeur** [comment, marked as answer]
If you have groups, then you can always include a self link for each of these groups. This allows a client to include a "See more…" link to explore more content related to that group. This is better than paginating within a group IMO.

If you're trying to paginate the feed itself that contains the groups, then that's technically allowed by OPDS 2.0 but not widely supported by clients *yet*.

**Reply by gotson** [reply]
> that's technically allowed by OPDS 2.0 but not widely supported by clients yet.

that is the source of the problem in my opinion. The specification should also provide guidance on what clients are supposed to implement.

**Reply by HadrienGardeur** [reply]
This is a tricky thing to deal with.

I've been involved in multiple standardization efforts and client conformance is always a slippery slope. You can't really go halfway if you go down that road.

I've seen at least three ways this can be handled:

- adding client conformance statements to the main specification
- creating a separate document for client conformance
- or limiting client conformance to a best practices document

**Reply by gotson** [reply]
There are plenty of statements like "Client MUST do this" in the existing specs. I don't see how that is a problem to add more, saying that acquisition feeds can be multiple pages, and that "Client MUST handle the paging appropriately and not just handle the first page".

**Reply by mickael-menu** [reply]
I agree that a MUST is warranted for the pagination of a publications/acquisition feed. It's not like an enhancement like search, if pagination is not implemented, the feed is broken.

However, I would forbid pagination at the global level for a mixed feed. It's not clear at all what happens in the user interface in this case, as we don't have a single grid to scroll.

Pagination *inside* a group could be technically possible, but I don't know any client implementing it. The typical "homepage" UI usually features a mixed feed that showcases collections with a subset of publications. If you want to see the whole publications (with pagination), you "expand" the collection. It's the UI implemented in most VOD and catalog services.

**Reply by mickael-menu** [reply]
Another benefit of doing that is the ability to further filter a collection using facets when expanded. When it's stuck within a mixed feed, its pretty limited.
